langs/befunge/strncpy.patch

```diff
@@ -0,0 +1,11 @@
+--- /lib/genx/genx.c
++++ /lib/genx/genx.c
+@@ -219,7 +219,7 @@
+ 		if ((w->status = growCollector(c, sl)) != GENX_SUCCESS)
+ 			return GENX_ALLOC_FAILED;
+ 
+-	strlcpy((char *) c->buf, (const char *) string, c->space);
++	strncpy((char *) c->buf, (const char *) string, c->space);
```

Comment:
This seems like a step backwards, isn't `strlcpy` the "good" one? What was the breakage we're working around?

Reply:
> What was the breakage we're working around?

`strlcpy`

```bash
12.32 -- Build files have been written to: /build
12.37 [  1%] Building C object CMakeFiles/cfunge.dir/lib/fungestring/funge_memchr.c.o
12.46 [  3%] Building C object CMakeFiles/cfunge.dir/lib/fungestring/funge_strchr.c.o
12.51 [  5%] Building C object CMakeFiles/cfunge.dir/lib/fungestring/funge_strlen.c.o
12.57 [  6%] Building C object CMakeFiles/cfunge.dir/lib/fungestring/funge_strstr.c.o
12.83 [  8%] Building C object CMakeFiles/cfunge.dir/lib/genx/charProps.c.o
13.48 [ 10%] Building C object CMakeFiles/cfunge.dir/lib/genx/genx.c.o
13.51 /lib/genx/genx.c: In function 'collectString':
13.51 /lib/genx/genx.c:222:9: error: implicit declaration of function 'strlcpy'; did you mean 'strncpy'? [-Wimplicit-function-declaration]
13.51   222 |         strlcpy((char *) c->buf, (const char *) string, c->space);
13.51       |         ^~~~~~~
13.51       |         strncpy
13.51 /lib/genx/genx.c:222:9: warning: nested extern declaration of 'strlcpy' [-Wnested-externs]
13.55 make[2]: *** [CMakeFiles/cfunge.dir/build.make:149: CMakeFiles/cfunge.dir/lib/genx/genx.c.o] Error 1
13.55 make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/cfunge.dir/all] Error 2
13.55 make: *** [Makefile:146: all] Error 2
```

Reply:
Ah is it just missing the header then?

Reply:
> Ah is it just missing the header then?

Yes, but I just found a better fix which you'll like more.